src/lists.rs

```diff
@@ -281,7 +285,7 @@ where
     let indent_str = &formatting.shape.indent.to_string(formatting.config);
     while let Some((i, item)) = iter.next() {
         let item = item.as_ref();
-        let inner_item = item.item.as_ref()?;
+        let inner_item: &String = item.item.as_ref().or_else(|err| Err(err.clone()))?;
```

Comment:
We needed to add an explicit type annotation here?

Reply:
No, it's unnecessary, and I'll remove this.